### DISCUSSION TITLE
How to prevent the plugin from removing unused imports? (UnusedImport)

### DISCUSSION THREAD

**Discussion by ErnestoBorio** [golang/vscode-go#1672] [Discussions]
I tried using these settings but none seems to work and I have no idea what setting is removing unused imports on save.
Any ideas?
It's ok that unused imports are errors and that go won't let me run or build with these kinds of errors, but I don't want it to delete the imports from the code because when I'm in the middle of coding I need to add the import statement first and then use its declared symbols, and it's silly to try and find workarounds like commenting the import line or not saving until I use it at least once.

    "go.lintOnSave": "off",
    "editor.formatOnSave": false,
    "go.buildOnSave": "off",
    "[go]": {
    	"editor.formatOnSave": false,
    }

**Comment by stamblerre** [comment]
You will need to use:

```
"[go]": {
	"editor.codeActionsOnSave": {
		"source.organizeImports": false,
	},
}
```

**Reply by ErnestoBorio** [reply]
Thanks!
